Types.hs

```diff
@@ -118,7 +119,40 @@ instance PathPiece StackageExecutable where
     fromPathPiece "stackage-setup.exe" = Just StackageWindowsExecutable
     fromPathPiece _ = Nothing
 
-type GhcMajorVersion = Text
+data GhcMajorVersion = GhcMajorVersion Int Int
+  deriving (Eq)
+
+ghcMajorVersionToText :: GhcMajorVersion -> Text
+ghcMajorVersionToText (GhcMajorVersion a b)
+  = pack (show a) <> "." <> pack (show b)
+
+ghcMajorVersionFromText :: Text -> Maybe GhcMajorVersion
+ghcMajorVersionFromText t = case T.splitOn "." t of
```

Comment:
Similarly, using the `Data.Text.Read` functions may be better here. Also, instead of returning a `Maybe`, it may be nice to use `MonadThrow`.